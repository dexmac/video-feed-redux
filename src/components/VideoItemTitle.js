import React from "react";
import PropTypes from "prop-types";

const VideoItemTitle = ({ title = "" }) => (
  <div className="video-item-title">{title}</div>
);

VideoItemTitle.propTypes = {
  title: PropTypes.string
};

export default VideoItemTitle;
