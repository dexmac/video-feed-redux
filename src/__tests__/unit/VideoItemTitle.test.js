import React from "react";
import VideoItemTitle from "../../components/VideoItemTitle";

describe("VideoItemTitle", () => {
  let props;
  let mountedVideoItemTitle;

  /**
   * Clearing props before each test
   */
  beforeEach(() => {
    props = {
      title: undefined
    };
    mountedVideoItemTitle = undefined;
  });

  /**
   * Shallow rendering vs. snapshot test
   */
  it("renders correctly && matches snapshot", () => {
    const tree = shallow(<VideoItemTitle />);
    expect(tree).toMatchSnapshot();
  });

  /**
   * Mounts the component using Enzyme if it is not mounted yet
   * @returns {object} - returns the mounted component
   */
  const videoItemTitle = () => {
    if (!mountedVideoItemTitle) {
      mountedVideoItemTitle = mount(<VideoItemTitle {...props} />);
    }
    return mountedVideoItemTitle;
  };

  describe("when `title` is defined", () => {
    beforeEach(() => {
      props.title = "myTitle";
    });

    it("sets the rendered `div` text to the same value as `title`", () => {
      const myDiv = videoItemTitle().find("div");
      expect(myDiv.text()).toBe(props.title);
    });
  });

  describe("when `title` is undefined", () => {
    beforeEach(() => {
      props.title = undefined;
    });

    it("sets the rendered `div` text to be an empty string", () => {
      const myDiv = videoItemTitle().find("div");
      expect(myDiv.text()).toBe("");
    });
  });
});
