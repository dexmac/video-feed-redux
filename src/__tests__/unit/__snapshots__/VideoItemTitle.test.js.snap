// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItemTitle renders correctly && matches snapshot 1`] = `
<div
  className="video-item-title"
/>
`;
